lib/Passes/CodeQualityStatCollector.cpp

```diff
@@ -62,15 +63,26 @@ CodeQualityStatCollector::run(llvm::Function &function,
   ConditionalComplexityVisitor complexity_visitor;
   for (auto &i : llvm::instructions(function)) {
     if (auto *int_to_ptr = llvm::dyn_cast<llvm::IntToPtrInst>(&i)) {
-      IntToPointerCasts++;
+      ++IntToPointerCasts;
     }
 
     if (auto *int_to_ptr = llvm::dyn_cast<llvm::PtrToIntInst>(&i)) {
-      PointerToIntCasts++;
+      ++PointerToIntCasts;
     }
 
+    if (auto *store_inst = llvm::dyn_cast<llvm::StoreInst>(&i)) {
+      if (store_inst->getPointerOperand()->getName() == kSymbolicSPName) {
+        ++AnvillStackPointers;
+      }
+    }
+
+    if (auto *load_inst = llvm::dyn_cast<llvm::LoadInst>(&i)) {
+      if (load_inst->getPointerOperand()->getName() == kSymbolicSPName) {
```

Comment:
This is too specific. A load inst may derive a pointer from `inttoptr( add (ptrtoint  anvill_sp) 15)`. You need to chase down an expression that built up a pointer to see if it references anvill_sp. There is infrastructure to do that https://github.com/lifting-bits/anvill/blob/dd89a2316c1d5ade3640aba5f727dc93c2605b67/include/anvill/Utils.h#L79 . We perhaps could also do a similar thing with this pass before anvill_pc removal to see if we missed pc relative references. Regardless of if we use that approach or some other approach we need a way to check if we missed globals as well